src/debugAdapter/goDebug.ts

```diff
@@ -321,21 +321,35 @@ class GoDebugSession extends DebugSession {
 		log('InitializeEvent');
 	}
 
+	protected findPathSeperator(path) {
+		if (/^(\w:[\\/]|\\\\)/.test(path)) return '\\';
+		return path.includes('/') ? '/' : '\\';
+	}
+
 	protected launchRequest(response: DebugProtocol.LaunchResponse, args: LaunchRequestArguments): void {
 		// Launch the Delve debugger on the program
+		let localPath = args.program;
 		let remotePath = args.remotePath || '';
 		let port = args.port || random(2000, 50000);
 		let host = args.host || '127.0.0.1';
 
 		if (remotePath.length > 0) {
-			this.localPathSeparator = args.program.includes('/') ? '/' : '\\';
```

Comment:
This fails for the (for me) common case where I share a project (and it's launch.json) and open it on different platforms and having a setting like `"program": "${workspaceRoot}/test"` because `${workspaceRoot}` has backslashes on Windows but slashes on Unix.